cli/src/utils/fileHandling.ts

```diff
@@ -130,3 +137,142 @@ export async function validateAndGetFileInfo(
 export function getFileExtension(filename: string): string {
   return extname(filename).toLowerCase();
 }
+
+export function detectFileType(
+  filePath: string
+): "text" | "image" | "pdf" | "audio" | "video" | "binary" {
+  const fileExtension = path.extname(filePath).toLowerCase();
+
+  // Handle TypeScript files specifically (mime library treats .ts as video)
+  if (fileExtension === ".ts") {
+    return "text";
+  }
+
+  // Check against known binary file extensions first
+  const binaryExtensions = new Set([
+    ".zip",
+    ".tar",
+    ".gz",
+    ".exe",
+    ".dll",
+    ".so",
+    ".class",
+    ".jar",
+    ".war",
+    ".7z",
+    ".doc",
+    ".docx",
+    ".xls",
+    ".xlsx",
+    ".ppt",
+    ".pptx",
+    ".odt",
+    ".ods",
+    ".odp",
+    ".bin",
+    ".dat",
+    ".obj",
+    ".o",
+    ".a",
+    ".lib",
+    ".wasm",
+    ".pyc",
+    ".pyo",
+  ]);
+
+  if (binaryExtensions.has(fileExtension)) {
+    return "binary";
+  }
+
+  // Use mime type detection for remaining files
+  const mimeType = mime.lookup(filePath);
+  if (!mimeType) {
+    return "text"; // Default to text if no mime type found
+  }
+
+  // Map mime types to file categories
+  const mimeTypeMap = {
+    "image/": "image",
+    "audio/": "audio",
+    "video/": "video",
+  } as const;
+
+  // Check for mime type prefixes
+  for (const [prefix, category] of Object.entries(mimeTypeMap)) {
+    if (mimeType.startsWith(prefix)) {
+      return category;
+    }
+  }
+
+  // Handle PDF specifically
+  if (mimeType === "application/pdf") {
+    return "pdf";
+  }
+
+  // Default fallback
+  return "text";
+}
+
+export async function processFile(
+  filePath: string,
+  offset: number = 0,
+  limit: number = MAX_LINES_TEXT_FILE
+) {
+  // check for existence
+  if (!fs.existsSync(filePath)) {
+    throw new Error(`File at ${filePath} does not exist.`);
+  }
+
+  // get stats, checks for file size and for whether it actually is a file and not a directory
+  const { fileType } = await validateAndGetFileInfo(filePath);
```

Comment:
this throws if the file is a directory
validateAndGetFileInfo should actually maybe also return a Result<>
at least we should circle that very statement with a try catch (and return a result<, error> here to the tool because processFile() should become a result<> according to comment above)